pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java

```diff
@@ -114,6 +113,21 @@ public static String parseStudyGuid(String path) {
         return studyGuid;
     }
 
+    /**
+     * Grab the study guid from the Admin URI path, e.g. `/pepper/v1/admin/studies/...`.
+     *
+     * @param path the admin URI path
+     * @return the study guid or null if the study could not be parsed
+     */
+    public static String parseAdminStudyGuid(String path) {
+        List<String> parts = SparkUtils.convertRouteToList(path);
+        String studyGuid = null;
+        if (parts.size() > ADMIN_STUDY_GUID_IDX && STUDIES_PATH_MARKER.equals(parts.get(ADMIN_STUDIES_MARKER_IDX))) {
+            studyGuid = parts.get(ADMIN_STUDY_GUID_IDX);
+        }
+        return studyGuid;
```

Comment:
Geez! Basic path parsing based on templates and extracting variables has been in every web framework I have seen except Spark.

Reply:
They have it, but last time I checked they don't expose it...